examples/ChatQnA/deploy/k8s_helm.md

```diff
@@ -0,0 +1,497 @@
+index based on decreasing order of the similarity score. The document
+corresponding to the returned index with the highest score is the most relevant
+document for the input query.
+```
+curl http://localhost:8808/rerank \
+    -X POST \
+    -d '{"query":"What is Deep Learning?", "texts": ["Deep Learning is not...", "Deep learning is..."]}' \
+    -H 'Content-Type: application/json'
+```
+
+Output is:  `[{"index":1,"score":0.9988041},{"index":0,"score":0.022948774}]`
+
+
+### TGI Service
+
+Use the following command to forward traffic from your local machine to the service running in the Kubernetes cluster:
+```bash
+kubectl port-forward svc/chatqna-tgi 9009:80
+```
+Follow the below steps in a different terminal.
+
+```
+curl http://localhost:9009/generate \
+  -X POST \
+  -d '{"inputs":"What is Deep Learning?","parameters":{"max_new_tokens":17, "do_sample": true}}' \
+  -H 'Content-Type: application/json'
+
+```
+
+TGI service generate text for the input prompt. Here is the expected result from TGI:
+
+```
+{"generated_text":"We have all heard the buzzword, but our understanding of it is still growing. It’s a sub-field of Machine Learning, and it’s the cornerstone of today’s Machine Learning breakthroughs.\n\nDeep Learning makes machines act more like humans through their ability to generalize from very large"}
+```
+
+**NOTE**: After launch the TGI, it takes few minutes for TGI server to load LLM model and warm up.
+
+If you get
+
+```
+curl: (7) Failed to connect to localhost port 8008 after 0 ms: Connection refused
+```
+
+and the log shows model warm up, please wait for a while and try it later.
+
+```
+2024-06-05T05:45:27.707509646Z 2024-06-05T05:45:27.707361Z  WARN text_generation_router: router/src/main.rs:357: `--revision` is not set
+2024-06-05T05:45:27.707539740Z 2024-06-05T05:45:27.707379Z  WARN text_generation_router: router/src/main.rs:358: We strongly advise to set it to a known supported commit.
+2024-06-05T05:45:27.852525522Z 2024-06-05T05:45:27.852437Z  INFO text_generation_router: router/src/main.rs:379: Serving revision bdd31cf498d13782cc7497cba5896996ce429f91 of model Intel/neural-chat-7b-v3-3
+2024-06-05T05:45:27.867833811Z 2024-06-05T05:45:27.867759Z  INFO text_generation_router: router/src/main.rs:221: Warming up model
+```
+
+### Dataprep Microservice (Advanced)
+
+Add Knowledge Base via HTTP Links:
+
+```
+curl -X POST "http://localhost:6007/v1/dataprep" \
+     -H "Content-Type: multipart/form-data" \
+     -F 'link_list=["https://opea.dev"]'
+```
+
+This command updates a knowledge base by submitting a list of HTTP links for processing.
+
+Also, you are able to get the file list that you uploaded:
+
+```
+curl -X POST "http://localhost:6007/v1/dataprep/get_file" \
+     -H "Content-Type: application/json"
+
+```
+
+To delete the file/link you uploaded you can use the following commands:
+
+#### Delete link
+```
+# The dataprep service will add a .txt postfix for link file
+
```

Comment:
Specified `file_path` value already has `.txt` suffix?

Reply:
@eero-t This is for a url which gets a `.txt`